### PR TITLE
pythonPackages.macholib: init at 1.16.3

### PR DESCRIPTION
**Pull request by Eveeifyeve**
## Description of changes

Adds https://github.com/ronaldoussoren/macholib which is required for other python binarys/applications eg. pyinstaller.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true` 
  - [x] `sandbox = false` 
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Eveeifyeve** [comment]
I am testing`nixpkgs-review` right now

**Comment by Eveeifyeve** [comment]
> I am testing`nixpkgs-review` right now.

I am going to wait until ofborg builds (aka going to asleep) it so it doesn't take a long time for testing.

File: pkgs/development/python-modules/macholib/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "macholib";
+  version = "1.16.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ronaldoussoren";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/macholib/default.nix:19]:
```suggestion
    repo = "macholib";
```

File: pkgs/development/python-modules/macholib/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  nativeCheckInputs = [
+    unittestCheckHook
+  ];
+
+  build-system = [ setuptools ];
+
+  preCheck = ''
+    export PATH="$PATH:$out/bin"
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/macholib/default.nix:41]:
```suggestion
  build-system = [ setuptools ];

  dependencies =
    [
      altgraph
    ]
    ++ lib.optionals (pythonOlder "3.11") [
      typing-extensions
    ];

  __darwinAllowLocalNetworking = true;

  nativeCheckInputs = [
    unittestCheckHook
  ];

  preCheck = ''
    export PATH="$PATH:$out/bin"
  '';
```

File: pkgs/development/python-modules/macholib/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  meta = with lib; {
+    description = "Analyze and edit Mach-O headers, the executable format used by Mac OS X.";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/macholib/default.nix:44]:
```suggestion
    description = "Analyze and edit Mach-O headers, the executable format used by Mac OS";
```

File: pkgs/development/python-modules/macholib/default.nix (hunk 2 above)
Comment (shelvacu) [reply pkgs/development/python-modules/macholib/default.nix:41]: Wait what? Why?

File: pkgs/development/python-modules/macholib/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  };
+
+  dependencies =
+    [
+      altgraph
+    ]
+    ++ lib.optionals (pythonOlder "3.11") [
+      typing-extensions
+    ];
+  __darwinAllowLocalNetworking = true;
```
Comment (shelvacu) [inline pkgs/development/python-modules/macholib/default.nix:31]: Why is this needed?

File: pkgs/development/python-modules/macholib/default.nix (hunk 3 above)
Comment (Eveeifyeve) [reply pkgs/development/python-modules/macholib/default.nix:44]: I just pulled this description from macholib.

File: pkgs/development/python-modules/macholib/default.nix (hunk 4 above)
Comment (Eveeifyeve) [reply pkgs/development/python-modules/macholib/default.nix:31]: It was needed for pypi but it's not needed anymore.

File: pkgs/development/python-modules/macholib/default.nix (hunk 2 above)
Comment (Eveeifyeve) [reply pkgs/development/python-modules/macholib/default.nix:41]: Does it matter which way stuff goes?

File: pkgs/development/python-modules/macholib/default.nix (hunk 3 above)
Comment (Eveeifyeve) [reply pkgs/development/python-modules/macholib/default.nix:44]: But yeah I guess this is fine.

File: pkgs/development/python-modules/macholib/default.nix (hunk 2 above)
Comment (Eveeifyeve) [reply pkgs/development/python-modules/macholib/default.nix]: I know the reason it's for readability so it's now fixed.

Comment (Eveeifyeve) [reply pkgs/development/python-modules/macholib/default.nix]: > Wait what? Why?

readability, SOLVED!

**Comment by Eveeifyeve** [comment]
If you need me to I could squash my commits?

**Review by shelvacu — APPROVED** [review]

**Comment by Eveeifyeve** [comment]
Squashed all of my commits.

**Comment by Eveeifyeve** [comment]
Result of `nixpkgs-review pr 344926` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.macholib</li>
    <li>python311Packages.macholib.dist</li>
    <li>python312Packages.macholib</li>
    <li>python312Packages.macholib.dist</li>
  </ul>
</details>

**Comment by Eveeifyeve** [comment]
Just added testing and fix available platforms.

**Comment by h7x4** [comment]
It would be great if you could split it into two commits, one for adding yourself to the maintainers list and one for adding the package. Sorry about the fuzz with the commit stuff

**Comment by Eveeifyeve** [comment]
I have done both testing with sandbox to true and with it true.

**Comment by Eveeifyeve** [comment]
> It would be great if you could split it into two commits, one for adding yourself to the maintainers list and one for adding the package. Sorry about the fuzz with the commit stuff

Well it's squashed how could I un-squash it?

**Comment by h7x4** [comment]
There are several ways, but the first way that comes to mind is `git reset --soft HEAD~1` to undo the current commit and keep the changes in the staging area, then make two new commits

**Comment by Eveeifyeve** [comment]
> There are several ways, but the first way that comes to mind is `git reset --soft HEAD~1` to undo the current commit and keep the changes in the staging area, then make two new commits



> There are several ways, but the first way that comes to mind is `git reset --soft HEAD~1` to undo the current commit and keep the changes in the staging area, then make two new commits

I asked in matrix as well but apparently there is a too called `git-revise` when you do `git revise -i <commit-hash>` it gives you an interactive persion to pick, sqrash, index & etc.

**Comment by h7x4** [comment]
That probably works just as well too :)

**Comment by Eveeifyeve** [comment]
> That probably works just as well too :)

There separated the commits.

**Review by shelvacu — APPROVED** [review]
